src/query/buildRQL.js

```diff
@@ -98,6 +99,17 @@ var buildRQL = {
         node = queryBuilder(Component, variables, RelayQL);
       } else {
```

Comment:
Let's be more explicit about when the new logic will take effect in order to avoid surprising behavior and have stronger invariants about the expected outcome:
- Only auto-inject the fragment when `queryBuilder` is a function with exactly zero arguments (anything else could be the user doing `(Component) => Relay.QL...`) - so make this a 3rd `else if` clause.
- When the above case is true, check the following conditions:

```javascript
if (GraphQL.isQuery(node) && node.fragments.length === 0) {
  if (!node.fields.every(field => field.isScalar())) {
    invariant(
      false,
      'RelayQL: Expected query `%s` to be empty. For example, use `node(id: $id)`, ' +
      'not `node(id: $id) { ... }`.'
      query.name
    );
  }
  var rootCall = ...
  ...
}
```

This prevent